exasol/nb_connector/extension_wrapper_common.py

```diff
@@ -53,30 +81,45 @@ def encapsulate_bucketfs_credentials(
     the connection object will instead turn the verification off. This is
     because there is no guarantee that the consumer of the connection object,
     i.e. a UDF, would have this custom CA list, and even if it would, its location
-    is unknown.
+    is unknown. This is only applicable for an On-Prem backend.
     """
 
-    bfs_host = conf.get(CKey.bfs_host_name, conf.get(CKey.db_host_name))
-    bfs_protocol = "https" if str_to_bool(conf, CKey.bfs_encryption, True) else "http"
-    bfs_dest = (
-        f"{bfs_protocol}://{bfs_host}:{conf.get(CKey.bfs_port)}/"
-        f"{conf.get(CKey.bfs_bucket)}/{path_in_bucket};{conf.get(CKey.bfs_service)}"
-    )
-    # TLS certificate verification option shall be provided in the fragment field.
-    verify: Optional[bool] = (False if conf.get(CKey.trusted_ca)
-                              else optional_str_to_bool(conf.get(CKey.cert_vld)))
-    if verify is not None:
-        bfs_dest += f'#{verify}'
+    backend = (bfs.path.StorageBackend.saas if get_backend(conf) == StorageBackend.saas
+               else bfs.path.StorageBackend.onprem)
+    if backend == bfs.path.StorageBackend.onprem:
+        host = conf.get(CKey.bfs_host_name, conf.get(CKey.db_host_name))
+        protocol = "https" if str_to_bool(conf, CKey.bfs_encryption, True) else "http"
+        url = f"{protocol}://{host}:{conf.get(CKey.bfs_port)}"
+        verify: Optional[bool] = (False if conf.get(CKey.trusted_ca)
+                                  else optional_str_to_bool(conf.get(CKey.cert_vld)))
+        conn_to = (f'"backend":"{backend.name}", '
```

Comment:
I am wondering if a dict and `", ".join(....)` would make sense here.
Or even a kwarg function. 

```python
def conn_to(**kwargs):
    return ", ".join(f'"{k}":"{v}"' for k,v in kwargs.items())

url = "https://exasol.com"
service_name = "bfs"
bucket_name = "my_bucket"
path = "a/b/c"

c = conn_to(
    url=url,
    service_name=service_name,
    bucket_name=bucket_name,
    path=path,
)
```
yields 
```
"url":"https://exasol.com", "service_name":"bfs", "bucket_name":"my_bucket", "path":"a/b/c"
```

Reply:
Very good idea, thanks.